src/game/Spells/Spell.cpp

```diff
@@ -2437,17 +2440,49 @@ void Spell::SetTargetMap(SpellEffectIndex effIndex, uint32 targetMode, bool targ
                                 break;
                             }
                         }
-
                     }
                 }
-             
+
                 UnitList tempAoeList;
-                MaNGOS::AnyFriendlyOrGroupMemberUnitInUnitRangeCheck u_check(m_caster, group, m_spellInfo, max_range);
-                MaNGOS::UnitListSearcher<MaNGOS::AnyFriendlyOrGroupMemberUnitInUnitRangeCheck> searcher(tempAoeList, u_check);
-                Cell::VisitAllObjects(m_caster, searcher, max_range);
+                // Getting spell casting distance
+                float minRadiusCaster = 0, maxRadiusTarget = 0;
+                GetChainJumpRange(m_spellInfo, effIndex, minRadiusCaster, maxRadiusTarget);
+
+                // if we have a group, we don't care about other units around
+                if (group) {
+                    for (GroupReference* itr = group->GetFirstMember(); itr != nullptr; itr = itr->next())
+                    {
+                        Player* member = itr->getSource();
+                        tempAoeList.push_back(member);
+                    }
+                } else {
+                    // Filling target map
+                    SpellNotifyPushType pushType = PUSH_TARGET_CENTER;
+                    FillAreaTargets(tempAoeList, maxRadiusTarget, cone, pushType, SPELL_TARGETS_ASSISTABLE);
+                }
+
+                // No targets. No need to process.
+                if (tempAoeList.empty())
+                    break;
+
+                // Allways remove currentTarget cause we already have it into tempUnitList
                 tempAoeList.erase(std::remove(tempAoeList.begin(), tempAoeList.end(), unitTarget), tempAoeList.end());
-                if (!tempAoeList.empty())
-                    tempUnitList.splice(tempUnitList.end(), tempAoeList);
+
+                if (minRadiusCaster)
```

Comment:
Chain heal has no min radius. The function fetcher has min radius because I added it there. The jump radius was configured based on patch notes (we are lucky to have jump radius specifically for chain heal from them). So this is entirely unnecessary and not how it works. The filtering is not done here, it is done in FilterTargetMap. So this block here is redundant.